buffer/src/main/java/io/netty/buffer/ReadableObject.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.buffer;
+
+import io.netty.util.ReferenceCounted;
+
+import java.io.IOException;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * An object that contains a readable region. For simplicity, extends {@link ReferenceCounted}.
+ */
+public interface ReadableObject extends ReferenceCounted {
+    /**
+     * Returns current read position in the object.
+     */
+    long readerPosition();
+
+    /**
+     * Sets the {@code readerPosition} of this object.
+     *
+     * @throws IndexOutOfBoundsException
+     *         if the specified {@code readerPosition} is
+     *            less than {@code 0} or
+     *            greater than {@link #readerLimit()}.
+     */
+    ReadableObject readerPosition(long readerPosition);
+
+    /**
+     * Gets the read position limit (exclusive). The value of {@code readerPosition} must always
+     * be less than or equal to this limit.
+     */
+    long readerLimit();
+
+    /**
+     * Returns {@code true} if and only if {@link #readableBytes()} > {@code 0}.
+     */
+    boolean isReadable();
+
+    /**
+     * Returns {@code true} if and only if this buffer contains equal to or more than the specified number of elements.
+     */
+    boolean isReadable(int size);
```

Comment:
@nmittler - `int` vs `long`.  Should we use `long` here to be consistent with the rest of the methods in this interface?